File: app/src/main/java/in/sudopk/vaishnavacalendar/DayCalendar.java
package in.sudopk.vaishnavacalendar;

import android.support.annotation.NonNull;

import java.util.List;

public class DayCalendar implements Comparable<DayCalendar> {
    private final int mDate;
    private final List<String> mEvents;

    public DayCalendar(final int date, final List<String> events) {
        mDate = date;
        mEvents = events;
    }

    public int getDate() {
        return mDate;
    }

    public List<String> getEvents() {
        return mEvents;
    }

    @Override
    public int compareTo(@NonNull final DayCalendar dayCalendar) {
        return getDate() - dayCalendar.getDate();
    }

    @Override
    public boolean equals(final Object obj) {
        return obj instanceof DayCalendar &&
                compareTo((DayCalendar) obj) == 0;
    }
}

File: app/src/main/java/in/sudopk/vaishnavacalendar/calendar/DayCalendar.kt
package `in`.sudopk.vaishnavacalendar.calendar

class DayCalendar(val date: Int, val events: List<String>) : Comparable<DayCalendar> {

    override fun compareTo(other: DayCalendar): Int {
        return date - other.date
    }

    override fun equals(other: Any?): Boolean {
        return other is DayCalendar && compareTo(other) == 0
    }

    override fun hashCode(): Int {
        return date;
    }
}
